api/cpp/esp-idf/slint/README.md

```diff
@@ -11,81 +11,110 @@ It has been tested on ESP32-S3 devices.
-Behind the scene, it will use `esp_lcd_panel_draw_bitmap` to render to the screen,
-and is base on [ESP LCD Touch](https://components.espressif.com/components/espressif/esp_lcd_touch) for the touch events.
-
-Use this platform implementation by instantiating it with a `esp_lcd_panel_handle_t`, an optional `esp_lcd_touch_handle_t`, and a pointer to one
-or two `slint::platform::Rgb565Pixel` frame buffers. Next, register the instance with the Slint run-time library by calling `slint_esp_init()`:
-
-```cpp
-#include "slint-esp.h"
-// ...
-
-
-// Initialize the panel and touch handle
-esp_lcd_panel_handle_t panel_handle = NULL;
-bsp_display_new(&bsp_disp_cfg, &panel_handle, &io_handle);
-bsp_display_backlight_on();
-esp_lcd_touch_handle_t touch_handle = NULL;
-bsp_touch_new(&bsp_touch_cfg, &touch_handle);
-
-// Allocate a frame buffer
-static std::vector<slint::platform::Rgb565Pixel> buffer(BSP_LCD_H_RES * BSP_LCD_V_RES);
-
-// initialize the Slint ESP backend
-slint_esp_init(slint::PhysicalSize({ BSP_LCD_H_RES, BSP_LCD_V_RES }), panel_handle,
-                                   touch_handle, buffer);
-
+import { VerticalBox, AboutSlint } from "std-widgets.slint";
+export component AppWindow inherits Window {
+    VerticalBox {
+        AboutSlint {}
+        Text {
+            text: "Hello World";
+            font-size: 18px;
+            horizontal-alignment: center;
+        }
+    }
+}
 ```
-
-Alternatively, you can implement your own sub-class of `slint::platform::Platform` to drive the screen and handle input events.
-
-Next, integrate your `.slint` files into the build by compiling them to C++ and linking them to your component:
-
+1. Edit `main/CMakeLists.txt` to adjust for the new `slint-hello-world.cpp`, add `slint` as required component,
+   and instruction the build system to compile `appwindow.slint` to `appwindow.h`. The file should look like this:
 ```cmake
-slint_target_sources(${COMPONENT_LIB} my_application_ui.slint)
-```
-
-Instantiate the Slint component by including the generated header file (`my_application_ui.h` in the above example), call `create()`
-on the generated class and `run()` to spin the event loop:
-
-```cpp
-#include "my_application_ui.h"
-
-/// ...
-
-auto my_application_ui = MainWindow::create();
-
-my_application_ui->run();
+idf_component_register(SRCS "slint-hello-world.cpp" INCLUDE_DIRS "." REQUIRES slint)
+slint_target_sources(${COMPONENT_LIB} appwindow.slint)
 ```
+1. Open the configuration editor with `idf.py menuconfig`:
+    1. Change the stack size under `Component config --> ESP System Settings --> Main task stack size` to `13584`.
+    2. Add support for C++ exceptions under `Compiler Options -> Enable C++ exceptions`.
+    3. Quit the editor with `Q` and save the configuration.
+2.  Build the project with `idf.py build`.
+3.  Connect your device, then flash and run it with `idf.py flash monitor`.
+4.  Observe Slint rendering "Hello World" on the screen 🎉.
```

Comment:
I observe blinking screen because it crashes at startup :-(

```
I (409) main_task: Started on CPU0
I (419) main_task: Calling app_main()
I (419) gpio: GPIO[4]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 
I (429) gpio: GPIO[48]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 
I (559) ESP-BOX: Setting LCD backlight: 100%

abort() was called at PC 0x42042ce3 on core 0
0x42042ce3: __cxxabiv1::__terminate(void (*)()) at /builds/idf/crosstool-NG/.build/xtensa-esp32s3-elf/src/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:48



Backtrace: 0x403759e2:0x3fcc54b0 0x4037aec5:0x3fcc54d0 0x40381be6:0x3fcc54f0 0x42042ce3:0x3fcc5560 0x42042d18:0x3fcc5580 0x42042df3:0x3fcc55a0 0x42042e7a:0x3fcc55c0 0x4200b249:0x3fcc55e0 0x4200b262:0x3fcc5600 0x4200b295:0x3fcc5620 0x4200ccb1:0x3fcc5640 0x42052a5f:0x3fcc56c0 0x4037d631:0x3fcc56f0
0x403759e2: panic_abort at /home/olivier/esp/esp-idf/components/esp_system/panic.c:452

0x4037aec5: esp_system_abort at /home/olivier/esp/esp-idf/components/esp_system/port/esp_system_chip.c:84

0x40381be6: abort at /home/olivier/esp/esp-idf/components/newlib/abort.c:38

0x42042ce3: __cxxabiv1::__terminate(void (*)()) at /builds/idf/crosstool-NG/.build/xtensa-esp32s3-elf/src/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:48

0x42042d18: std::terminate() at /builds/idf/crosstool-NG/.build/xtensa-esp32s3-elf/src/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:58

0x42042df3: __cxa_throw at /builds/idf/crosstool-NG/.build/xtensa-esp32s3-elf/src/gcc/libstdc++-v3/libsupc++/eh_throw.cc:98

0x42042e7a: operator new(unsigned int) at /builds/idf/crosstool-NG/.build/xtensa-esp32s3-elf/src/gcc/libstdc++-v3/libsupc++/new_op.cc:54

0x4200b249: std::__new_allocator<slint::platform::Rgb565Pixel>::allocate(unsigned int, void const*) at /home/olivier/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/include/c++/12.2.0/bits/new_allocator.h:137

0x4200b262: std::allocator<slint::platform::Rgb565Pixel>::allocate(unsigned int) at /home/olivier/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/include/c++/12.2.0/bits/allocator.h:188
 (inlined by) std::allocator_traits<std::allocator<slint::platform::Rgb565Pixel> >::allocate(std::allocator<slint::platform::Rgb565Pixel>&, unsigned int) at /home/olivier/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/include/c++/12.2.0/bits/alloc_traits.h:464
 (inlined by) std::_Vector_base<slint::platform::Rgb565Pixel, std::allocator<slint::platform::Rgb565Pixel> >::_M_allocate(unsigned int) at /home/olivier/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/include/c++/12.2.0/bits/stl_vector.h:378
 (inlined by) std::_Vector_base<slint::platform::Rgb565Pixel, std::allocator<slint::platform::Rgb565Pixel> >::_M_create_storage(unsigned int) at /home/olivier/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/include/c++/12.2.0/bits/stl_vector.h:395

0x4200b295: std::_Vector_base<slint::platform::Rgb565Pixel, std::allocator<slint::platform::Rgb565Pixel> >::_Vector_base(unsigned int, std::allocator<slint::platform::Rgb565Pixel> const&) at /home/olivier/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/include/c++/12.2.0/bits/stl_vector.h:332
 (inlined by) std::vector<slint::platform::Rgb565Pixel, std::allocator<slint::platform::Rgb565Pixel> >::vector(unsigned int, std::allocator<slint::platform::Rgb565Pixel> const&) at /home/olivier/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/include/c++/12.2.0/bits/stl_vector.h:552

0x4200ccb1: app_main at /home/olivier/esp/slint-hello-world/main/slint-hello-world.cpp:31 (discriminator 2)

0x42052a5f: main_task at /home/olivier/esp/esp-idf/components/freertos/app_startup.c:208 (discriminator 13)

0x4037d631: vPortTaskWrapper at /home/olivier/esp/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/port.c:162


```

I'll try to debug that

Reply:
We need to enable the RAM
`Component config → ESP PSRAM  → Support for external, SPI-connected RAM`
`Component config → ESP PSRAM → Support for external, SPI-connected RAM → SPI RAM config → Mode (QUAD/OCT) of SPI RAM chip in use` set `Octal Mode PSRAM`
`Component config → ESP PSRAM → Support for external, SPI-connected RAM → SPI RAM config → Set RAM clock speed` set `80MHz clock speed`

Reply:
Uhhh, odd, I didn't need that on the ESP32S3-USB-OTG. I'll try to find there's anything in the ESP-IDF docs that helps with a good default for devices. Otherwise we may have to tell people to use an sdkconfig default from an example...